File: CivitasRuby/lib/dado.rb
# encoding:utf-8

module Civitas
  class Dado
    include Singleton
    
    def initialize
      @random = 1
      @ultimo_resultado = 1
      @debug = false
      @@salida_carcel = 5
    end
    
      
      
    def tirar
      if(!debug)
        @ultimo_resultado = rand(6)+1
      else
        @ultimo_resultado = 1
      end
      
      return @ultimo_resultado
    end    
    
    
    
    def salgo_de_la_carcel
      @valor = tirar()
      @salir = false
      if(valor >= 5)
        salir = true
      end
      
      return salir
    end
    
    
    
    def quien_empieza (n)
      @primer_jugador = rand(n-1)+1
      return @primer_jugador
    end
    
    
    attr_writer :debug
    attr_reader :ultimo_resultado
    
  end
end

File: CivitasJava/src/Civitas/Dado.java
package Civitas;
import java.util.ArrayList;
import java.util.Random;

public class Dado {
    
    private static final Dado instance = new Dado();
    private int ultimoResultado;
    private Random random = new Random();
    private boolean debug;
    private static int SalidaCarcel;
    
    
    static public Dado getInstance() {
        return instance;
    }
    
    
    private Dado() {
        SalidaCarcel = 5;
        debug = false;
        ultimoResultado = 1;
    }
    
    
    int tirar() {
        if (!debug)
            ultimoResultado = random.nextInt(6)+1;
        else
            ultimoResultado = 1;
            
        return ultimoResultado;
    }
    
    
    boolean salgoDeLaCarcel (){
        int valor;
        boolean salir = false;
        valor = tirar();
        if (valor >= 5)
            salir = true;
        
        return salir;
    }
    
    
    int quienEmpieza (int n){
        int primerJugador = 1;
        primerJugador = random.nextInt(n) + 1;
        return primerJugador;
    }
    
    
    public void setDebug(boolean d) {
        this.debug = d;
    }
    

    int getUltimoResultado() {
        return ultimoResultado;
    }

    public boolean isDebug() {
        return debug;
    }
    
}
